### PR TITLE
Update to cilium 1.14

### PR DESCRIPTION
**Pull request by fiunchinho**
This is what the upgrade to cilium 1.14 would look like. I'm targeting a temporary branch in our cilium fork. Please review the changes, and when that's approved we can merge that branch into main, then merge this PR.

### CODE REVIEW

**Review by whites11 — APPROVED** [review]
it is pretty much impossible to review the PR in a meaningful way, so let's rely on testing

**Comment by fiunchinho** [comment]
> it is pretty much impossible to review the PR in a meaningful way, so let's rely on testing

Famous last words

**Review by njuettner — APPROVED** [review]

File: helm/cilium/Chart.yaml [hunk 1]
```diff
@@ -2,10 +2,10 @@ apiVersion: v2
 name: cilium
 displayName: Cilium
 home: https://cilium.io/
-version: 0.17.0
-appVersion: 1.13.6
+version: 1.14.3
```
Comment (AndiDog) [inline helm/cilium/Chart.yaml:5]: The `version` field should be updated automatically to the app version on release

**Review by AndiDog — APPROVED** [review]
One nit

**Pull request merged**